5.1.vinaigrette.py

```diff
@@ -0,0 +1,48 @@
+from datetime import datetime as dt
+import random
+
+def string_to_date(str_date):
```

Comment:
This code is not clean at all, consider:

1. Separate code to small functions- 
https://softwareengineering.stackexchange.com/questions/308108/when-is-it-appropriate-to-make-a-separate-function-when-there-will-only-ever-be

2.  Change variable names - https://workat.tech/machine-coding/tutorial/writing-meaningful-variable-names-clean-code-za4m83tiesy0 - variable names, abbreviation reference is missing